safehttp/mux_test.go

```diff
@@ -597,3 +659,41 @@ func TestMuxHandlerReturnsNotWritten(t *testing.T) {
 		t.Errorf(`response body got: %q want: ""`, got)
 	}
 }
+
+type onErrorWriteInterceptor struct{}
+
+func (onErrorWriteInterceptor) Before(w *safehttp.ResponseWriter, _ *safehttp.IncomingRequest, _ safehttp.InterceptorConfig) safehttp.Result {
+	return w.WriteError(safehttp.StatusBadRequest)
+}
+
+func (onErrorWriteInterceptor) Commit(_ *safehttp.ResponseWriter, _ *safehttp.IncomingRequest, _ safehttp.Response, _ safehttp.InterceptorConfig) safehttp.Result {
+	return safehttp.Result{}
+}
+
+func (onErrorWriteInterceptor) OnError(w *safehttp.ResponseWriter, _ *safehttp.IncomingRequest, _ safehttp.Response, _ safehttp.InterceptorConfig) safehttp.Result {
+	w.Header().Set("header", "bad")
+	return w.WriteError(safehttp.StatusInsufficientStorage)
+}
+
+func TestMuxInterceptorWriteInOnError(t *testing.T) {
+	b := &strings.Builder{}
+	rw := safehttptest.NewTestResponseWriter(b)
+
+	req := httptest.NewRequest(safehttp.MethodGet, "http://foo.com/bar", nil)
+
+	mb := &safehttp.ServeMuxConfig{}
+	mb.Intercept(onErrorWriteInterceptor{})
+
+	registeredHandler := safehttp.HandlerFunc(func(w *safehttp.ResponseWriter, r *safehttp.IncomingRequest) safehttp.Result {
+		return w.Write(safehtml.HTMLEscaped("<h1>Hello World!</h1>"))
+	})
+	mb.Handle("/bar", safehttp.MethodGet, registeredHandler)
+
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf(`mb.Mux().ServeHTTP(rw, req): expected panic`)
+		}
+	}()
```

Comment:
```suggestion
	defer func() {
		if r := recover(); r != nil {
			return
		}
		t.Errorf(`mb.Mux().ServeHTTP(rw, req): expected panic`)
	}()
```

SuperNit: Reading a recover with a `== nil` condition might be surprising for some users.